cdbot/cogs/admin.py

```diff
@@ -89,6 +92,32 @@ async def on_member_join(self, member: Member):
             # assign placeholder nickname
             await member.edit(nick=PLACEHOLDER_NICKNAME)
 
+    @Cog.listener()
+    async def on_message(self, message):
+        def check(m):
+            return m.author == message.author and (datetime.datetime.utcnow() - m.created_at).seconds < 4
+
+        def who(m):
+            return m.author == message.author
+
+        if not message.author.bot:
+            # Checks if message contains banned word.
+            if any([word in message.content for word in BANNED_WORDS]):
+                await message.delete()
+                await message.channel.send(f"{message.author.mention} Watch your language!", delete_after=5)
+            # Checks if message contains banned domain.
+            if any([word in message.content for word in BANNED_DOMAINS]):
+                await message.delete()
+                await message.channel.send(f"{message.author.mention} No invite links!", delete_after=5)
+            # Checks if message contains mass mention.
+            if len(message.mentions) > 8:
+                await message.delete()
+                await message.channel.send(f"{message.author.mention} Don't spam mentions!", delete_after=5)
```

Comment:
At this point, I'm getting a sense of DRY. It may be better to encapsulate the checks, then have a single call to `message.delete()` and to `message.challenge.send()` after the encapsulated checks.